### PR TITLE
Global code search support

### PR DESCRIPTION
**Pull request by lunny**
This PR added a code search UI on explore. When indexer is enabled, both guest user and login user could use the feature.  Guest user will only search public codes and login user could also search private codes he could access.
image

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

**Review by sapk — APPROVED** [review]

**Comment by ghost** [comment]
Thank you for this wonderful functionality guys. Can you release it on 1.4.0?

**Comment by jonasfranz** [comment]
@giudon 1.4.0 is already release candidate that means that no features will be added.

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -169,9 +169,12 @@ repos = Repositories
 users = Users
 organizations = Organizations
 search = Search
+code = Code
 repo_no_results = No matching repositories have been found.
 user_no_results = No matching users have been found.
 org_no_results = No matching organizations have been found.
+code_no_results = No matching code have been found.
```
Comment (jonasfranz) [inline options/locale/locale_en-US.ini]: „code have been found“ should be changed to „code has been changed“

File: models/repo_list.go [hunk 2]
```diff
@@ -249,3 +249,28 @@ func SearchRepositoryByName(opts *SearchRepoOptions) (RepositoryList, int64, err
 
 	return repos, count, nil
 }
+
+// FindUserAccessiableRepoIDs find all accessiable repositories' ID by user's id
+func FindUserAccessiableRepoIDs(userID int64) ([]int64, error) {
```
Comment (jonasfranz) [inline models/repo_list.go]: does this include organization repos that could be accessed by the user?

**Review by jonasfranz — CHANGES_REQUESTED** [review]

File: models/repo_list.go (hunk 2 above)
Comment (mbger) [reply models/repo_list.go]: And a typo in **accessible**?

File: options/locale/locale_en-US.ini (hunk 1 above)
Comment (lafriks) [reply options/locale/locale_en-US.ini]: Why changed?

Comment (lunny) [reply options/locale/locale_en-US.ini]: Maybe I should use `No matching codes have been found.`

File: models/repo_list.go (hunk 2 above)
Comment (lunny) [reply models/repo_list.go]: @JonasFranzDEV yes, included.
@mbger done.

File: options/locale/locale_en-US.ini (hunk 1 above)
Comment (lunny) [reply options/locale/locale_en-US.ini]: @JonasFranzDEV @lafriks done.

**Comment by lunny** [comment]
And I also add the UnitTypeCode check when login user is not admin.

**Review by appleboy — APPROVED** [review]

**Review by jonasfranz — APPROVED** [review]

**Review by sapk — APPROVED** [review]

**Pull request merged**